pennylane_lightning/src/Gates.cpp

```diff
@@ -74,6 +102,12 @@ const vector<CplxType> Pennylane::YGate::matrix{
     0, -IMAG,
     IMAG, 0 };
 
+void Pennylane::YGate::applyKernel(CplxType* state, vector<size_t>& indices, vector<CplxType>& v) {
+    v[0] = state[indices[0]];
+    state[indices[0]] = -IMAG * state[indices[1]];
+    state[indices[1]] = IMAG * v[0];
```

Comment:
How come we need to update `v[0]` here (and not for `PauliX` for example)? Seems like it's used as a temp for the swap, but it would also have effect afterwards, correct?

Reply:
`v` is really just scratch space. If we move the outer loop in, we can avoid allocating `v` altogether, except in the base implementation. The requirement to pass `v` is one reason why I'm not entirely happy with this interface.

Reply:
Got it! 

> we can avoid allocating v altogether, except in the base implementation

This would be preferable as just creating `v` would result in an overhead, correct? Could we branch of at the beginning of the outer loop based on the type of operation?

Reply:
The allocation for `v` is really insignificant in the grand scheme of things, so I don't think branching is necessary. The first rule of optimisation is to not over-optimise, and I don't think this ever becomes a bottleneck, so its more a question of code clarity as to which API we adopt.